server/runtime/src/main/java/org/infinispan/server/state/ServerStateManagerImpl.java

```diff
@@ -0,0 +1,278 @@
+      return cache.putAsync(new ScopedState(CONNECTOR_IPFILTER_SCOPE, name), new IpFilterRules()).thenApply(v -> null);
+   }
+
+   private void updateLocalIgnoredCaches(IgnoredCaches ignored) {
+      if (ignored != null) {
+         synchronized (this) {
+            this.ignored.caches.clear();
+            this.ignored.caches.addAll(ignored.caches);
+            hasIgnores = !this.ignored.caches.isEmpty();
+         }
+      }
+   }
+
+   @Listener(observation = Listener.Observation.POST)
+   private final class IgnoredCachesListener {
+      @CacheEntryCreated
+      public void created(CacheEntryCreatedEvent<ScopedState, IgnoredCaches> e) {
+         if (!e.isOriginLocal()) {
+            updateLocalIgnoredCaches(e.getValue());
+         }
+      }
+
+      @CacheEntryModified
+      public void modified(CacheEntryModifiedEvent<ScopedState, IgnoredCaches> e) {
+         if (!e.isOriginLocal()) {
+            updateLocalIgnoredCaches(e.getValue());
+         }
+      }
+   }
+
+   @Listener(observation = Listener.Observation.POST)
+   private final class ConnectorStateListener {
+      @CacheEntryCreated
+      public void created(CacheEntryCreatedEvent<ScopedState, Boolean> e) {
+         // stop the connector
+         server.getProtocolServers().get(e.getKey().getName()).getTransport().stop();
+      }
+
+      @CacheEntryRemoved
+      public void removed(CacheEntryRemovedEvent<ScopedState, Boolean> e) {
+         // start the connector
+         server.getProtocolServers().get(e.getKey().getName()).getTransport().start();
+      }
+   }
+
+   @Listener(observation = Listener.Observation.POST)
+   private final class ConnectorIpFilterListener {
+      @CacheEntryCreated
+      @CacheEntryModified
+      public CompletionStage<Void> modified(CacheEntryEvent<ScopedState, IpFilterRules> e) {
+         String connector = e.getKey().getName();
+         ProtocolServer protocolServer = server.getProtocolServers().get(connector);
+         List<IpSubnetFilterRule> rules = new ArrayList<>(e.getValue().rules.size());
+                  for (IpFilterRule rule : e.getValue().rules) {
+            rules.add(new IpSubnetFilterRule(rule.cidr, IpFilterRuleType.valueOf(rule.type)));
+         }
+         protocolServer.getConfiguration().ipFilter().rules(rules);
+         Transport transport = protocolServer.getTransport();
+         if (transport != null) {
+            return transport.applyIpFilterRules(rules).thenApply(v -> {
+               Server.log.connectorIpFilterSet(connector, rules);
+               return v;
+            });
+         } else {
+            return CompletableFutures.completedExceptionFuture(Server.log.noConnectorTransport(connector));
+         }
+      }
+
+      @CacheEntryRemoved
+      public void removed(CacheEntryRemovedEvent<ScopedState, IpFilterRules> e) {
+         String connector = e.getKey().getName();
+         server.getProtocolServers().get(connector).getConfiguration().ipFilter().rules().clear();
```

Comment:
can be removed. we no longer remove the key from the cache.

Reply:
Done